src/codegen/compile_classical.ml

```diff
@@ -12330,10 +12261,7 @@ and compile_prim_invocation (env : E.t) ae p es at =
 
   | ComponentPrim (_, component_name, function_name, arg_types, return_type), es ->
     if not !Flags.wasm_components then
-        begin
-          Wasm.Sexpr.print 80 (Wasm.Sexpr.Atom ("Wasm components support not enabled, use -wasm-components"));
-          exit 0
-        end;
+      failwith "Wasm components support not enabled, use -wasm-components";
```

Comment:
nit: How about reverting this change (and undeleting the corresponding test for the flag), so that we're sure that the flag is working?  It seems to me that `failwith` is a bit misleading here (as the error message says that one has found a bug in the compiler, which is not accurate in this case), yet having some feedback about about the failure is useful.  (no strong opinion, if you prefer as-is, fine for me)